pkg/minikube/download/download_test.go

```diff
@@ -183,41 +183,60 @@ func testImageToCache(t *testing.T) {
 	}
 }
 
+// Validates that preload existence checks correctly caches values retrieved by remote checks.
 // Validates that preload existence checks correctly caches values retrieved by remote checks.
 func testPreloadExistsCaching(t *testing.T) {
 	checkCache = func(_ string) (fs.FileInfo, error) {
 		return nil, fmt.Errorf("cache not found")
 	}
 	doesPreloadExist := false
-	checkCalled := false
-	checkRemotePreloadExists = func(_, _ string) bool {
-		checkCalled = true
+	gcsCheckCalls := 0
+	ghCheckCalls := 0
+	originalGCSCheck := checkRemotePreloadExistsGCS
+	originalGHCheck := checkRemotePreloadExistsGitHub
```

Comment:
The common name of this is savedXxx - but are we modifying global here? why?

Reply:
I dont undrestnad the comment, can you plz elaborate